src/main/resources/org/sagebionetworks/web/client/widget/table/TableListWidgetViewImpl.ui.xml

```diff
@@ -22,7 +22,7 @@
 					<tr:SortableTableHeaderImpl
 						ui:field="nameColumnHeader" text="Name" />
 					<t:TableHeader width="100px"
-						addStyleNames="visible-md visible-lg">
+						addStyleNames="visible-sm visible-md visible-lg">
```

Comment:
There was plenty of room to show the new "Type" column in the table list, so I made it visible at small widths. This the header change, but I also made the change on the table data.